operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java

```diff
@@ -625,10 +626,20 @@ public boolean wasInspectionRun() {
 
     private boolean isShouldContinue() {
       DomainPresenceInfo cachedInfo = getExistingDomainPresenceInfo(getNamespace(), getDomainUid());
+      String existingError = Optional.ofNullable(liveInfo)
+          .map(DomainPresenceInfo::getDomain)
+          .map(Domain::getStatus)
+          .map(DomainStatus::getMessage)
+          .orElse(null);
       if (cachedInfo == null || cachedInfo.getDomain() == null) {
         return true;
       } else if (isCachedInfoNewer(liveInfo, cachedInfo)) {
         return false;  // we have already cached this
+      } else if (existingError != null && existingError.startsWith("MII Fatal Error")
+          && !isSpecChanged(liveInfo, cachedInfo)) {
```

Comment:
what if the spec is changed and the change is causing a fatal error?

Reply:
Good point, I think I can remove the check

Reply:
Well, if I don't put it there, then after fixing the underlying error and changing the spec will not run the introspector ever again and essentially have to delete the domain and retry.

Reply:
If the spec change is causing a fatal error, and the operator continues retry, will the isSpecChanged() returns false in the next retry?  In other word, does the operator update the current info on each retry? If so, I think you are good.

Reply:
Yes, that's the idea.